Mathlib/Analysis/Normed/Unbundled/SpectralNorm.lean

```diff
@@ -0,0 +1,666 @@
+        (algebraMap E (normalClosure K E (AlgebraicClosure E)) g) =
+      spectralNorm K L x :=
+  h_map ▸ spectralNorm.eq_of_normalClosure' E g
+
+variable (y : L)
+
+open Real
+
+/-- `spectralNorm K L (0 : L) = 0`. -/
+theorem spectralNorm_zero : spectralNorm K L (0 : L) = 0 := by
+  unfold spectralNorm spectralValue spectralValueTerms
+  rw [minpoly.zero, natDegree_X]
+  convert ciSup_const using 2
+  · ext m
+    by_cases hm : m < 1
+    · rw [if_pos hm, Nat.lt_one_iff.mp hm, Nat.cast_one, Nat.cast_zero, sub_zero, div_one,
+        rpow_one, coeff_X_zero, norm_zero]
+    · rw [if_neg hm]
+  · infer_instance
+
+/-- `spectralNorm K L y` is nonnegative. -/
+theorem spectralNorm_nonneg (y : L) : 0 ≤ spectralNorm K L y :=
+  le_ciSup_of_le (spectralValueTerms_bddAbove (minpoly K y)) 0 (spectralValueTerms_nonneg _ 0)
+
+/-- `spectralNorm K L y` is positive if `y ≠ 0`. -/
+theorem spectralNorm_zero_lt {y : L} (hy : y ≠ 0) (hy_alg : IsAlgebraic K y) :
+    0 < spectralNorm K L y := by
+  apply lt_of_le_of_ne (spectralNorm_nonneg _)
+  rw [spectralNorm, ne_eq, eq_comm, spectralValue_eq_zero_iff (minpoly.monic hy_alg.isIntegral)]
+  have h0 : coeff (minpoly K y) 0 ≠ 0 := minpoly.coeff_zero_ne_zero hy_alg.isIntegral hy
+  intro h
+  have h0' : (minpoly K y).coeff 0 = 0 := by
+    rw [h, coeff_X_pow, if_neg (ne_of_lt (minpoly.natDegree_pos hy_alg.isIntegral))]
+  exact h0 h0'
+
+/-- If `spectralNorm K L x = 0`, then `x = 0`. -/
+theorem eq_zero_of_map_spectralNorm_eq_zero {x : L} (hx : spectralNorm K L x = 0)
+    (hx_alg : IsAlgebraic K x) : x = 0 := by
+  by_contra h0
+  exact (ne_of_gt (spectralNorm_zero_lt h0 hx_alg)) hx
+
+/-- If `f` is a power-multiplicative `K`-algebra norm on `L` with `f 1 ≤ 1`, then `f`
+  is bounded above by `spectralNorm K L`. -/
+theorem norm_le_spectralNorm {f : AlgebraNorm K L} (hf_pm : IsPowMul f)
+    (hf_na : IsNonarchimedean f) (hf1 : f 1 ≤ 1) {x : L} (hx_alg : IsAlgebraic K x) :
+    f x ≤ spectralNorm K L x :=
+  norm_root_le_spectralValue hf_pm hf_na hf1 (minpoly.monic hx_alg.isIntegral)
+    (by rw [minpoly.aeval])
+
+/-- The `K`-algebra automorphisms of `L` are isometries with respect to the spectral norm. -/
+theorem spectralNorm_eq_of_equiv (σ : L ≃ₐ[K] L) (x : L) :
+    spectralNorm K L x = spectralNorm K L (σ x) := by
+  simp only [spectralNorm, minpoly.algEquiv_eq]
+
+-- We first assume that the extension is finite and normal
+
+section FiniteNormal
+
+variable (K L) [h_fin : FiniteDimensional K L] [hn : Normal K L]
+
+/--
+If `L/K` is finite and normal, then `spectralNorm K L x = supr (λ (σ : L ≃ₐ[K] L), f (σ x))`. -/
+theorem spectralNorm_eq_iSup_of_finiteDimensional_normal
+    {f : AlgebraNorm K L} (hf_pm : IsPowMul f) (hf_na : IsNonarchimedean f)
+    (hf_ext : ∀ (x : K), f (algebraMap K L x) = ‖x‖) (x : L) :
+    spectralNorm K L x = ⨆ σ : L ≃ₐ[K] L, f (σ x) := by
+  classical
+  have hf1 : f 1 ≤ 1 := by
+    rw [← (algebraMap K L).map_one, hf_ext]
+    simp [nnnorm_one, NNReal.coe_one, le_refl]
```

Comment:
```suggestion
  have hf1 : f 1 = 1 := by
    rw [← (algebraMap K L).map_one, hf_ext]
    simp [nnnorm_one, NNReal.coe_one]
```
You prove it's = 1 later on, why bother proving it's <= 1 now? Just prove it's = 1 now.